adserver/forms.py

```diff
@@ -313,6 +314,64 @@ class Meta:
         }
 
 
+class FlightCreateForm(forms.ModelForm):
+
+    """Create a new flight for this advertiser."""
+
+    def __init__(self, *args, **kwargs):
+        """Set the flight form helper and initial data for creating a flight."""
+        if "advertiser" in kwargs:
+            self.advertiser = kwargs.pop("advertiser")
+        else:
+            raise RuntimeError("'advertiser' is required for the flight create form")
+
+        super().__init__(*args, **kwargs)
+
+        self.fields["campaign"].queryset = Campaign.objects.filter(
+            advertiser=self.advertiser
+        )
+        self.helper = FormHelper()
+        self.helper.layout = Layout(
+            Fieldset(
+                "",
+                "name",
+                "campaign",
+                css_class="my-3",
+            ),
+            Submit("submit", _("Create flight")),
+            HTML(
+                "<p class='form-text small text-muted'>"
+                + str(_("You will be able to edit details on the next screen"))
+                + "</p>"
+            ),
+        )
+
+    def generate_slug(self):
+        campaign_slug = self.instance.campaign.slug
+        slug = slugify(self.instance.name)
+        if not slug.startswith(campaign_slug):
+            slug = slugify(f"{campaign_slug}-{slug}")
+
+        while Flight.objects.filter(slug=slug).exists():
+            random_char = get_random_string(1)
+            slug = slugify(f"{slug}{random_char}")
```

Comment:
```suggestion
            slug = slugify(f"{slug}-{random_char}")
```

To make it a bit cleaner?